src/reaktoro_pse/reaktoro_block.py

```diff
@@ -0,0 +1,952 @@
+            This generally should not be needed""",
+        ),
+    )
+
+    def build(self):
+        super().build()
+        """ configure state"""
+        if self.config.build_speciation_block:
+            """create spectitation block and then property block"""
+            self.speciation_block = Block()
+            self.build_rkt_state(self.speciation_block, speciation_block=True)
+            self.build_rkt_inputs(self.speciation_block, speciation_block=True)
+            self.build_rkt_outputs(self.speciation_block, speciation_block=True)
+            self.build_rkt_jacobian(self.speciation_block)
+            self.build_rkt_solver(self.speciation_block, speciation_block=True)
+            self.build_gray_box(self.speciation_block)
+
+            self.build_rkt_state(
+                self,
+                speciation_block=False,
+                speciation_block_built=True,
+                input_composition=self.speciation_block.outputs,
+            )
+            self.build_rkt_inputs(
+                self, speciation_block=False, speciation_block_built=True
+            )
+            self.build_rkt_outputs(self, speciation_block=False)
+            self.build_rkt_jacobian(self)
+            self.build_rkt_solver(self, speciation_block=False)
+            self.build_gray_box(self)
+        else:
+            """create property block only"""
+            self.build_rkt_state(self)
+            self.build_rkt_inputs(self)
+            self.build_rkt_outputs(self)
+            self.build_rkt_jacobian(self)
+            self.build_rkt_solver(self)
+            self.build_gray_box(self)
+
+    def build_rkt_state(
+        self,
+        block,
+        speciation_block=False,
+        speciation_block_built=False,
+        input_composition=None,
+    ):
+        """this will buld our rkt state specified block.
+        The keyword arguments are for automatic configuration of speciation and property blocks
+
+        Keywords:
+        block -- pyomo block to build the model on
+        speciation_block -- sets to build speciation block which by default does not
+            add mineral or gas phases (unless configured to do so), and only outputs speciesAmount from the rktModel
+        speciation_block_built -- this should only be True if speciation block was built, in this case the model
+            will build user requested outputs, not provide user supplied pH and disable charge neutrality
+        input_composition -- if alternate input composition should be used rather one provided by config (used to pass in
+            speciesAmount output from speciation_block)
+
+        """
+
+        """ defining which indexes shold be indexed 
+         if block is not indexed, index is None 
+         if specific input is set to be not Indexed the index will be set to None"""
+
+        index = self.index()
+        composition_indexed = index
```

Comment:
An inline comment as to why these are needed might be helpful - my initial reaction was that they were unnecessary until I read the next part.

Reply:
done